glossary.yml

```
@@ -1749,6 +1749,23 @@
     def: >
       An unambiguous name of the form `package::thing`.
 
+- slug: function
+  ref:
+    - functions
+    - command
+  en:
+    term: "function"
+    def: >
+      A code block which gathers a sequence of operations into a whole, preserving it for ongoing use
+      by a defining tasks that take a set of inputs and optional arguments to return expected outputs.
```

```suggestion
      by defining a set of tasks that takes zero or more required and optional arguments as inputs and return expected outputs (return values), if any.
```

Elsewhere in this glossary we (read: I) have defined all of the input values passed to a function as arguments, which can be either required or optional, and with the understanding that a function can take required arguments, optional arguments, a mixture of the two, or no arguments at all. Consequently, this wording in English feels inconsistent.

Return values are also optional in some languages (Python, for example, has an optional `return` statement, and will implicitly return `None` if no `return` statement or return value are given).

@zkamvar  Thoughts on this altered English wording? (Note, I've also fixed a typo.) Also, the tense of 'to take' here depends on whether a group of objects is treated as a single thing, or several, and the US/K environments I live in are conflicted here.

---

Thanks that's helpful to know (as of course I'm new to the project). If it's possible to be consistent with other terms that would be ideal!

This is a more difficult one to define that I'd initially expected. Different programming languages have slightly different terminology for these concepts. @masamiy has actually raised similar issues with the Japanese version with the added confusion of English loanwords or Chinese characters. In hindsight, a bilingual contribution was ambitious. Thanks for checking it.